pkg/workflow/engine_test.go

```diff
@@ -0,0 +1,130 @@
+package workflow
+
+import (
+	"fmt"
+	"net/url"
+	"testing"
+
+	"github.com/snyk/go-application-framework/pkg/configuration"
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_EngineBasics(t *testing.T) {
+	configuration := configuration.New()
```

Comment:
Can we maybe divide this function to make it shorter and clearer?

Reply:
I can move out the callback functions which blow the test up that much.